tests/console/nvme_checks.pm

```diff
@@ -0,0 +1,113 @@
+    foreach (@nvm_partitions) {
+        assert_script_run("test -b $_ || echo 'nvm is not a block device'");
+    }
+}
+
+sub _exercise_nvme_commands_and_validate_output {
+    my $nvm_test_data = shift;
+    # send Identify Admin Command to the NVMe controller.
+    assert_script_run("nvme id-ctrl -H /dev/$nvm_test_data->{nvm_char_device} > id_ctrl_output");
+    record_info "Identify Admin Command Output", script_output("cat id_ctrl_output");
+    my $MODE       = '\/dev\/' . $nvm_test_data->{nvm_char_device};
+    my $SN         = $nvm_test_data->{sn};
+    my $MODEL      = $nvm_test_data->{model};
+    my $NAMESPACE  = $nvm_test_data->{namespace_count};
+    my $DISK_SIZE  = $nvm_test_data->{nvm_disk_size};
+    my $BLOCK_SIZE = $nvm_test_data->{nvm_block_size};
+    my $list_regex = qr/${MODE}\s+$SN\s+${MODEL}\s+${NAMESPACE}\s+${DISK_SIZE}\s+GB.*\d+\s+GB\s+${BLOCK_SIZE}/;
+    validate_script_output("nvme list",                                                                     sub { $list_regex });
+    validate_script_output("nvme list-ns /dev/$nvm_test_data->{nvm_char_device} | awk -F ':' '{print $2}'", sub { /$nvm_test_data->{nvm_ns}/ });
+    # the details of this namespace
+    # check LBA Format fields
+    # ms = Metadata Size, 0 means metadata is not supported
+    # lbads = LBA Data Size, in terms of a power of two, so LBA data size is 2⁹ = 512 bytes
+    # rp = Relative Performance, the lower the better
+    validate_script_output("nvme id-ns /dev/$nvm_test_data->{nvm_disk} --namespace-id=$nvm_test_data->{nvm_ns}", sub { m/lbaf\s+0\s+:\s+ms:0\s+lbads:9\s+rp:0\s\(in use\)/ });
+    # show registers
+    my $ret_code = script_run("nvme show-regs -H /dev/$nvm_test_data->{nvm_disk}");
+    record_soft_failure "bsc#1175217 - not permitted shared memory access to get registers" if ($ret_code > 0);
+}
+
+sub _issue_read_command_and_compare_with_dd {
+    my $args = shift;
+    assert_script_run("dd if=/dev/nvme0n1 of=lba.0.dd bs=512 count=1");
+    # read LBA 0 through nvme utility
+    assert_script_run("nvme read /dev/nvme0n1 --start-block=0 --block-count=0 --data-size=512 --data=lba.0.read");
+    # Read is opcode 0x02, we are sending command to namespace 0x1, reading 512 bytes
+    # code word 10 and 11 (cdw10 and cdw11) specifies the start of reading in LBA blocks,
+    # and the Bits 15:00 of code word 12 (cdw12) is number of blocks to read (set to 0 to indicate 1 block will be read).
+    assert_script_run("nvme io-passthru /dev/nvme0 --opcode=0x02 --namespace-id=0x1 --data-len=512 --read --cdw10=0 --cdw11=0 --cdw12=0 -b > lba.0.io");
+    assert_script_run("cmp lba.0.dd lba.0.read");
+    assert_script_run("cmp lba.0.dd lba.0.io", fail_message => "The Submission and Completion queue for Admin commands differs from I/O commands");
+}
+
+sub run {
+    select_console 'root-console';
+    my $testdata = get_test_suite_data();
+    record_soft_failure "bsc#1172866 - nvme-cli is not installed" unless script_run("rpm -qa nvme-cli");
```

Comment:
you could wrap the soft failure in another private function (it is done like that in many files in the repo).
By the way this script_run seems to return always '0' and the zypper in should be done only when the package is not installed. Let's check with zypper as well, for example:
```
my $ret = zypper_call('se -x -i nvme-cli', exitcode => [0, 104]);
    if ($ret == 104) {
    record_soft_failure ...
    zypper_call('in ...
    ...
```